tests/tasks/test_bgpstatemonitortask.py

```diff
@@ -54,9 +54,185 @@ async def test_task_logs_missing_information(self, snmpsim, snmp_test_port, capl
         assert f"router {device.name} misses BGP variables" in caplog.text
 
     @pytest.mark.asyncio
-    async def test_peer_admin_status_changing_to_stop_while_peer_state_is_not_established_should_create_event_general(
-        self, snmpsim, snmp_test_port
-    ):
+    async def test_external_reset_general_creates_event(self, snmpsim, snmp_test_port):
+        device = PollDevice(
+            name=DEVICE_NAME,
+            address=DEVICE_ADDRESS,
+            community="general-bgp-external-reset",
+            port=snmp_test_port,
+        )
+        state = ZinoState()
+        task = BgpStateMonitorTask(device, state)
```

Comment:
We did discuss a bit about reemoving redundancy from the tests with conclusion that the tests being very transparent in what is being done being better than fewer lines. But I think its fine to move these lines to its own Fixture. The only thing from these being used later on in the test is the `task` object, so making a `general_external_reset_task` fixture (and similar for the other snmprec files) would make the tests easier to read by removing complexity, without removing any relevant context. the `DEVICE_NAME`, `DEVICE_ADDRESS` etc being used to define the PollDevice is largely irrelevant for these tests.

The fixture name would still make it clear what snmprec file is being used, and that we are operating on a `BgpMonitorTask` object.

Reply:
Thanks for giving me that push and sending me down the pytest rabbit hole! I was able to strongly simplify the tests!